tortoisebeacon/tortoise_beacon.go

```diff
@@ -359,6 +376,10 @@ func (tb *TortoiseBeacon) handleEpoch(ctx context.Context, epoch types.EpochID)
 
 		return
 	}
+	if !tb.sync.IsSynced(ctx) {
```

Comment:
or you can register a channel to be notified when the node turned synced. see `RegisterChForSynced`. 
tho this does not notify you again if the node goes out of sync, i think it serves your purpose here.

Reply:
that would be better. but if node goes out of sync again i will need to re-register. and for that, i still need a pointer